### PR TITLE
OCT-1598 rewards api sanity test

### PR DESCRIPTION
**Pull request by MateuszStoleckiGLM**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: backend/tests/api-e2e/test_api_rewards.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    res = client.get_estimated_budget_by_days(300, 10000000000000000000000)
+    three_hundreds_days_budget_estimation = int(res["budget"])
+    assert three_hundreds_days_budget_estimation > 0
+    assert three_hundreds_days_budget_estimation > two_hundreds_days_budget_estimation
+
+    # write merkle root for withdrawals
+    vault_core.confirm_withdrawals()
+
+    while not vault.is_merkle_root_set(STARTING_EPOCH):
+        time.sleep(1)
```
Comment (kgarbacinski) [inline backend/tests/api-e2e/test_api_rewards.py:96]: is this time.sleep(1) necessary here? it always makes the api tests running longer that may be pain in the ass in our pipelines

**Review by kgarbacinski — APPROVED** [review]
LGTM

File: backend/tests/api-e2e/test_api_rewards.py (hunk 1 above)
Comment (MateuszStoleckiGLM) [reply backend/tests/api-e2e/test_api_rewards.py:96]: It is necessary to sync. We are using it twice in the code right now but I will look for improvements once it is more frequent.

Comment (MateuszStoleckiGLM) [reply backend/tests/api-e2e/test_api_rewards.py:96]: > is this time.sleep(1) necessary here? it always makes the api tests running longer that may be pain in the ass in our pipelines
It is necessary to sync. We are using it twice in the code right now but I will look for improvements once it is more frequent.

**Review by paulperegud — APPROVED** [review]

**Pull request merged**